### ISSUE TITLE
Add support for sampleText custom parameter

### ISSUE THREAD

**Issue by chrissimpkins** [googlefonts/ufo2ft#370]
I am working on a project with non-Latin script builds that do not include full basic Latin glyph sets.  When a user installs the fonts with Font Book on macOS, they see a string of glyphs that are included in the font and a string of tofu. We would like to define this installer string with glyphs that are known to be present in the fonts.

We are building with fontmake and added the `sampleText` (= UFO spec `openTypeNameSampleText`) custom parameter to the glyphs source files but this parameter does not appear to be supported in glyphsLib.  Thanks for considering support here!

**Comment by belluzj** [comment]
I tried this transfer issue button, and it already had "ufo2ft" as the only suggestion, spooky :)

**Comment by anthrotype** [comment]
ok, it appears we only set nameIDs from 0 to 17:

https://github.com/googlefonts/ufo2ft/blob/c76448f0607d63f6d80672ccbd5ee4f2e0f194aa/Lib/ufo2ft/outlineCompiler.py#L371-L388

it should be trivial to add support for the missing ones.

for reference, the list of `openTypeName*` attribuets is under "OpenType Name Table Fields" at:
http://unifiedfontobject.org/versions/ufo3/fontinfo.plist/

**Comment by belluzj** [comment]
Another (new) one that could be supported as well is the name id 25 for variable fonts https://docs.microsoft.com/en-us/typography/opentype/spec/name#nid25

**Comment by madig** [comment]
Yeah but 25 should be set on variable fonts, so it should be a DS level parameter.

**Comment by anthrotype** [comment]
For `Variations PostScript Name Prefix` (nameID=25) maybe we could specify a new `public.openTypeVariationsPostScriptNamePrefix` (what a mouthful) key to be added to the designspace document's lib.

**Comment by anthrotype** [comment]
the `openTypeNameCompatibleFullName` (nameID=18) has this fallback defined in fontInfoData.py:

https://github.com/googlefonts/ufo2ft/blob/c76448f0607d63f6d80672ccbd5ee4f2e0f194aa/Lib/ufo2ft/fontInfoData.py#L204-L214

however this is not used anywhere in outlineCompiler, only in some unit tests.

I think it doesn't make sense to have a fallback for a legacy and optional nameID like 18 (used only by old mac apps, and not commonly found in modern fonts). If the attribute is not set in UFO, ufo2ft should not set it in the TTFont, period.

do you guys agree? I'm gonna remove that.

**Comment by madig** [comment]
https://github.com/fonttools/fonttools/issues/1741 :)

**Comment by chrissimpkins** [comment]
Sorry for the delay in getting back to you @belluzj.  Thank you all very much for adding this support!  Very much appreciated!

**Comment by anthrotype** [comment]
Although, it looks like simply setting nameID=19 doesn't seem to work in FontBook's preview... I'm on Catalina (10.15.3).
Note that ufo2ft by default uses platformID=3 (windows) platEncID=1 (unicode) langID=0x409 (English) for name records strings (unless one uses custom `openTypeNameRecords` info attribute).

Can anybody confirm whether they had any success in setting the Sample Text string with Apple FontBook?

**Comment by anthrotype** [comment]
@chrissimpkins It seems like if I use `langID="0x40d"` (the [Windows language ID for Hebrew](https://docs.microsoft.com/en-us/typography/opentype/spec/name#windows-language-ids)) then it works in FontBook:

```diff
-    <namerecord nameID="19" platformID="3" platEncID="1" langID="0x409">
+    <namerecord nameID="19" platformID="3" platEncID="1" langID="0x40d">
       עטלף אבק נס דרך מזגן שהתפוצץ כי חם
     </namerecord>
```

![image](https://user-images.githubusercontent.com/6939968/77553557-dda17e00-6eac-11ea-89d2-c5f4ceeaed81.png)

With UFO3 one can set [openTypeNameRecords](http://unifiedfontobject.org/versions/ufo3/fontinfo.plist/) with explicit custom platformID, platEncID, langID triplets. 

How do you do that with Glyphs.app? Can one set a `sampleText` custom parameter with a custom plat/enc/lang?

**Comment by anthrotype** [comment]
there's a `Name Table Entry` custom parameter that allows to set custom name string in this format:

`<nameID> <platformID> <encID> <langID>; <nameString>`

https://glyphsapp.com/tutorials/naming

unfortunately glyphsLib does not support that yet.

**Comment by anthrotype** [comment]
setting `Name Table Entry` custom parameter from Glyphs.app to this string

`19 3 1 0x40D; עטלף אבק נס דרך מזגן שהתפוצץ כי חם`

generates the correct name record, when exporting from Glyphs.app

```xml
    <namerecord nameID="19" platformID="3" platEncID="1" langID="0x40d">
      עטלף אבק נס דרך מזגן שהתפוצץ כי חם
    </namerecord
```

to support this we need glyphsLib to translate from/to UFO3 `openTypeNameRecords` and Glyphs.app's `Name Table Entry` custom parameter.

@belluzj put a TODO here:
https://github.com/googlefonts/glyphsLib/blob/09e4cb6c07005fc46ebcfc88063ee0a90d6d504c/Lib/glyphsLib/builder/custom_params.py#L485-L501

**Comment by anthrotype** [comment]
related glyphsLib issue https://github.com/googlefonts/glyphsLib/issues/383

**Comment by chrissimpkins** [comment]
Thank you very much Cosimo!  I'll look into a workaround that edits the name table following compiles until this is available.  I really appreciate all of your help. 👍

**Comment by khaledhosny** [comment]
I use fontBuilder and it generates the following entry
```
    <namerecord nameID="19" platformID="3" platEncID="1" langID="0x409">        
      صف خلق خود كمثل الشمس إذ بزغت يحظى الضجيع بها نجلاء معطار                 
    </namerecord>
```

and it works in FontBook.

**Comment by anthrotype** [comment]
Yeah, we do the same, the only difference is in the contents of the string, your contains Arabic text, Chris' has Hebrew. No idea why FontBook would treat them differently.
What macOS version have you got?

**Comment by khaledhosny** [comment]
10.15.3, but indeed if I change the text to Hebrew FontBook no longer shows it and displays a List of basic Arabic characters instead. Oddly enough, if I go to the info tab, the sample entry at the end shows the Hebrew text.
